proto/src/protobuf/erased.rs

```diff
@@ -0,0 +1,40 @@
+use alloc::boxed::Box;
+use core::convert::{Into as CoreInto, TryFrom as CoreTryFrom};
+
+mod sealed {
```

Comment:
We can remove the `sealed` module entirely. It basically defines two "marker traits", and "marks" all types that satisfy the constraints:
+ `SealedInto<T>` for `U: Clone + CoreInto<T>`
+ `SealedTryFrom<T>` for `U: CoreTryFrom<T>`
These markers are then used as supertraits to our `Into` and `TryFrom`.

We see the same constraints being applied to our blanket implementations. So, the supertraits to our `Into` and `TryFrom` can simply be removed; the blanket implementation takes care of the constraints.

TLDR: if we remove the `sealed` module, the `sealed::SealedInto<T>` supertrait on `Into`, and the `sealed::SealedTryFrom<T>` supertrait on `TryFrom`, everything compiled and we've reduced the complexity of this module.

Reply:
The sealed traits signal that they are not supposed to be implemented by the user. I think it's nice to have them because we provide blanket impls for these traits.

Reply:
But what does having them provide that my suggestion doesn't? I understand that they're not supposed to be implemented by the user; I just don't see why they're useful at all.

>TLDR: if we remove the sealed module, the sealed::SealedInto<T> supertrait on Into, and the sealed::SealedTryFrom<T> supertrait on TryFrom, everything compiles and we've reduced the complexity of this module.